### DISCUSSION TITLE
giza package depends on a cairo that does not have X11 installed

### DISCUSSION THREAD

**Discussion by danieljprice** [Homebrew/discussions#919] [Tap maintenance and Homebrew development]
I'm not sure on the policy here, so wanted to raise a discussion. It is great to see a giza package officially added to Homebrew:

Formula/giza.rb

However there is a problem - the official version correctly makes giza depend on libX11, but the problem is that the official Cairo package in homebrew (Formula/cairo.rb) arrives with NO X11 installed, because of an earlier policy issue about not wanting to require Xquartz. This makes the giza package -- a plotting library with interactive features -- rather useless, as only the non-interactive drivers work.

My current workaround is to continue to maintain my own taps for this at https://github.com/danieljprice/homebrew-all/, including a customised Cairo tap that turns on the X11 drivers.

I was wondering if this might be solved more easily, now that the :x11 is replaced by lower-level dependencies (like libx11, libxcb etc)? Namely would it be possible for the Cairo package to now contain the x11 driver by default? Or is there still some policy against this?

**Comment by SMillerDev** [comment]
The issue was always with dependencies that weren't build/controlled by brew. If you can make it work with just formula that's a welcome addition.

**Reply by danieljprice** [reply]
I think this should be true now for the Cairo package, e.g.:

https://github.com/danieljprice/homebrew-all/blob/master/cairo-x11.rb

I guess the request would be for the Cairo package to be updated to something like what is in the above (ie. with dependencies on libx11, libxcb etc and with the  --enable-xlib=yes and similar flags turned on in the Cairo build). Updating this in the Cairo package would solve the issue with giza.

**Reply by SMillerDev** [reply]
Go ahead, I'll happily review a pull request to make software better for homebrew users.
